regulations/views/preamble.py

```diff
@@ -172,21 +178,69 @@ def make_preamble_toc(nodes, depth=1, max_depth=3):
     ]
 
 
-def section_navigation(full_id, toc):
+def preamble_section_navigation(full_id, preamble_toc, cfr_toc):
     nav = {'previous': None, 'next': None, 'page_type': 'preamble-section'}
-    for idx, sect in enumerate(toc):
+    for idx, sect in enumerate(preamble_toc):
         if sect.full_id == full_id:
             if idx > 0:
-                nav['previous'] = NavItem.from_section(toc[idx - 1])
-            if idx < len(toc) - 1:
-                nav['next'] = NavItem.from_section(toc[idx + 1])
+                nav['previous'] = NavItem.from_preamble_section(
+                    preamble_toc[idx - 1])
+            if idx < len(preamble_toc) - 1:
+                nav['next'] = NavItem.from_preamble_section(
+                    preamble_toc[idx + 1])
+            elif cfr_toc:
+                nav['next'] = NavItem.from_object(cfr_toc[0])
+    return nav
+
+
+def cfr_section_navigation(label_id, preamble_toc, cfr_toc):
```

Comment:
Doesn't block this PR, but there's a lot of overlap between `cfr_section_navigation` and `preamble_section_navigation`, hey? Why not generate your list of `NavItems` once and step through the combined list in a single function?